src/main.ts

```diff
@@ -279,7 +279,14 @@ async function archive(bufPath: string, inputs: Inputs): Promise<Result> {
   for (const label of inputs.archive_labels) {
     args.push("--label", label);
   }
-  return run(bufPath, args);
+  const result = await run(bufPath, args);
```

Comment:
How does the backend behave if we try to archive multiple labels at once and one of the labels doesn't exist. Does the whole RPC atomically fail? If so, I think we want to best effort archive as many labels as possible so I think we might want to make one archive command per label that needs to be archived.

Reply:
I don't understand why we would want one to succeed? Whats the use case?

Reply:
not sure I understand your question, can chat tomorrow if you want

Reply:
If one of the labels doesn't exist it will fail atomically. The default, and the most common case, would be to archive the one label from the delete event. So if multiple tags got deleted at once each event would fire and archive them in separate CI events. If a user overrides and wants multiple events I think failing if one doesn't exist is correct and we shouldn't try to run them one by one. Happy to chat through.

Reply:
Ok, so the difference here only matters if the user configured the GHA with one or more extra labels for archiving. Thinking about it now, why do we allow them to configure extra labels for archiving? Not sure that actually makes sense.

Reply:
We allow configuring more than one label for push, so makes sense if they have a multiple label pattern.

Reply:
> We allow configuring more than one label for push, so makes sense if they have a multiple label pattern.

Ok, yeah makes sense.

> If a user overrides and wants multiple events I think failing if one doesn't exist it is correct to fail and we shouldn't try to run them one by one.

Can chat about this live today, but I guess I view archiving state in the BSR as a cleanup task. If the archive task has been configured with let's say 2 labels, and one is valid but the other isn't, wouldn't you as a user prefer that the valid one gets archived even thought the other one doesn't exist?

Whether to mark the job as failed or not if this happens is a separate question, but I would be fine not marking the job has failed because we have asserted some state (i.e. the labels in question have been archived or don't exist)

Reply:
Passing-by comment: All APIs in v1/v1beta1 are atomic in its implementation, either all of them succeed or all of them fail, no partial success.

It's not explicitly documented in the `ArchiveLabels` RPC, but it would be weird having some RPCs that support partial success and others that are atomic. If we go down the path of partial success, we'd probably need a way to communicate in the response which ones were archived and which ones were not found, and it would open the gate to allow partial success for other RPCs like `Get*` or `Delete*`.

Reply:
Correct, I was not suggesting changing the API, I am suggesting changing the implementation here to loop and make multiple calls.